ColorPair.cpp

```diff
@@ -0,0 +1,45 @@
+#include "ColorPair.h"
+
+const char* MajorColorNames[] = { "White", "Red", "Black", "Yellow", "Violet" };
+int numberOfMajorColors = sizeof(MajorColorNames) / sizeof(MajorColorNames[0]);
+const char* MinorColorNames[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+int numberOfMinorColors = sizeof(MinorColorNames) / sizeof(MinorColorNames[0]);
+
+std::string TelCoColorCoder::ColorPair::ToString() {
+	std::string colorPairStr = MajorColorNames[majorColor];
+	colorPairStr += ", ";
+	colorPairStr += MinorColorNames[minorColor];
+	return colorPairStr;
+}
+
+TelCoColorCoder::ColorPair TelCoColorCoder::GetColorFromPairNumber(int pairNumber) {
+	int zeroBasedPairNumber = pairNumber - 1;
+	MajorColor majorColor = (MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+	MinorColor minorColor = (MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+	return TelCoColorCoder::ColorPair(majorColor, minorColor);
+}
+
+int TelCoColorCoder::GetPairNumberFromColor(MajorColor major, MinorColor minor) {
+	return major * numberOfMinorColors + minor + 1;
+}
+
+void TelCoColorCoder::PrintColorCodeManual()

```

Comment:
Always consider writing new functionality in a separate (new) file.
It helps in recognizing the functionality from the filename as well.